### PR TITLE
Only store the assets in the assets container

### PR DESCRIPTION
**Pull request by thewilkybarkid**
This should reduce the size, as we don't need to distribute the tools required to build the assets.

### CODE REVIEW

**Comment by giorgiosironi** [comment]
> CI which currently uses it for Node

If the `Dockerfile.assets` is split into `Dockerfile.assets-builder` and `Dockerfile.assets`, `Dockerfile.ci` can then extend the former?

**Comment by thewilkybarkid** [comment]
The `assets` container is ok, and is down from 355MB to 2MB.

File: Dockerfile.assets [hunk 1]
```diff
@@ -58,3 +23,12 @@ COPY --chown=elife:elife source/ ${PROJECT_FOLDER}/source
 
 COPY --chown=elife:elife assets/sass ${PROJECT_FOLDER}/assets/sass
 RUN gulp --environment ${environment} generateStyles
+
+FROM busybox:1.28.3
```
Comment (giorgiosironi) [inline Dockerfile.assets]: If this is going to be purely files without executables, you could try [`FROM scratch`](https://hub.docker.com/_/scratch/)

Comment (giorgiosironi) [reply Dockerfile.assets]: Although `docker-compose.yml` (being oriented to services) needs a command to execute so easier to have `sh` around

Comment (thewilkybarkid) [reply Dockerfile.assets]: Yeah, tried that first. Wasn't sure what to use as `command`.

File: docker-compose.yml [hunk 2]
```diff
@@ -30,8 +33,15 @@ services:
         command: /bin/bash
         depends_on:
             - assets
+            - node
             - ui
             - selenium
+    node:
```
Comment (giorgiosironi) [inline docker-compose.yml]: `node` should be at the top probably

File: Dockerfile.assets [hunk 3]
```diff
@@ -58,3 +23,12 @@ COPY --chown=elife:elife source/ ${PROJECT_FOLDER}/source
 
 COPY --chown=elife:elife assets/sass ${PROJECT_FOLDER}/assets/sass
 RUN gulp --environment ${environment} generateStyles
+
```
Comment (giorgiosironi) [inline Dockerfile.assets]: shouldn't everything before this go into the `node` image? Then the `ci` image can use it (e.g. the `gulpfile.js` absence which the makes the build fail)

File: Dockerfile.assets [hunk 4]
```diff
@@ -1,60 +1,11 @@
...
-
-COPY --chown=elife:elife assets/js/ ${PROJECT_FOLDER}/assets/js
-RUN gulp --environment ${environment} js
-
-COPY --chown=elife:elife source/ ${PROJECT_FOLDER}/source
 
-COPY --chown=elife:elife assets/sass ${PROJECT_FOLDER}/assets/sass
-RUN gulp --environment ${environment} generateStyles
+COPY --from=assets-builder \
+    --chown=root:root \
```
Comment (giorgiosironi) [inline Dockerfile.assets]: `--chown=root:root` should be the default of `COPY`

**Pull request merged**